### ISSUE TITLE
test_udp spits a DeprecationWarning regarding flushErrors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2154]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| PenguinOfDoom reported|
|-|-|
|Trac ID|trac#2154|
|Type|defect|
|Created|2006-10-06 21:55:53Z|

Here's the output (Linux):

```
pahan@dunce:~/projects/iocp_Twisted$ trial twisted.test.test_udp.UDPTestCase.testDatagramReceivedError
Running 1 tests.

/home/pahan/projects/iocp_Twisted/twisted/trial/runner.py:243: exceptions.DeprecationWarning: log.startKeepingErrors is deprecated since Twisted 2.5
twisted.test.test_udp
  UDPTestCase
    testDatagramReceivedError ... 
/home/pahan/projects/iocp_Twisted/twisted/test/test_udp.py:449: exceptions.DeprecationWarning: log.flushErrors is deprecated since Twisted 2.5. If you need to flush errors from within a unittest, use TestCase.flushLoggedErrors instead.
                                         [OK]

-------------------------------------------------------------------------------
Ran 1 tests in 0.012s

PASSED (successes=1)
```

<details><summary>Searchable metadata</summary>

```
trac-id__2154 2154
type__defect defect
reporter__PenguinOfDoom PenguinOfDoom
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__tests tests
time__1160171753000000 1160171753000000
changetime__1167068911000000 1167068911000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Fixed in test_udp-flush-errors-2154

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_3"></a>|@jerub commented|
|-|-|

Excellent, thanks to PenguinOfDoom for reporting this simple-to-fix problem, and thanks to exarkun for spending the time to write the patch.

Please fix these pyflakes warnings. One is introduced by this patch, the other has been there for longer.
```
$ pyflakes twisted/test/test_udp.py
twisted/test/test_udp.py:8: 'address' imported but unused
twisted/test/test_udp.py:9: 'log' imported but unused
```

Fix the import lines and merge this branch please.

Merry Christmas. :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [19085]) Merge test_udp-flush-errors-2154

Author: exarkun
Reviewer: jerub
Fixes [#2154](https://github.com/twisted/twisted/issues/2154)

Switch from deprecated twisted.python.log.flushErrors to TestCase.flushLoggedErrors.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
